frb_codegen/src/generator/rust/mod.rs

```diff
@@ -445,7 +472,7 @@ impl<'a> Generator<'a> {
                 body.map(|body| {
                     format!(
                         "impl Wire2Api<{api}> for {}{} {{
-                            fn wire2api(self) -> {api} {{
+                            fn wire2api(self) -> Result<{api}, &'static str> {{
```

Comment:
Hmm so `wire2api` is now fall-able... Is it worth it to add this complexity, or is there any way to keep it non-fallable?

Reply:
So far this is the best way.
without a lot of complex validation code on the dart side.
without unnecessary deallocation functions (reuse of wire2api).
And the fact that wire2api can return an error is quite logical for me.

validation of nested structures, panic or error should be on the executor also imposes restrictions

Reply:
(also see alternative solution mentioned below)